### PR TITLE
13 create sidebar

### PR DESCRIPTION
**Pull request by TrebuhD**
US#13 - As a user, I want to be able to see the sidebar with link categories.

Remove the ‘protected’ route, move its children up.

Rewrite sidebar using ember-sidebars: https://github.com/ef4/ember-sidebars

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,25 @@
...
+
+.idea/linkie-webapp.iml
+
+.idea/misc.xml
+
+.idea/modules.xml
+
+.idea/vcs.xml
+
+.idea/workspace.xml
```
Comment (Divoolej) [inline .gitignore]: Why are you being a Bad Person? :(

File: .travis.yml [hunk 2]
```diff
@@ -10,12 +10,11 @@ cache:
...
-  - "npm config set spin false"
-  - "npm install -g npm@^2"
+  - npm config set spin false
+  - npm install -g bower
+  - npm install phantomjs-prebuilt
 
 install:
-  - npm install -g bower
   - npm install
   - bower install
```
Comment (Divoolej) [inline .travis.yml:20]: Wait, what - do we use Travis? I didn't know that. We have no tests whatsoever at the moment, so.. was this autogenerated or sth?

Comment (TrebuhD) [inline .travis.yml:20]: Indeed it was

2016-04-25 9:31 GMT+02:00 Wojciech Olejnik notifications@github.com:

File: README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ You will need the following things properly installed on your computer.
 * [Git](http://git-scm.com/)
 * [Node.js](http://nodejs.org/) (with NPM)
 * [Bower](http://bower.io/)
-* [Ember CLI](http://www.ember-cli.com/)
+* [Ember CLI](http://ember-cli.com/)
```
Comment (Divoolej) [inline README.md:13]: This change deserves it's own Pull Request!

File: app/styles/sidebar.scss [hunk 4]
```diff
@@ -0,0 +1,120 @@
+// sidebar
+.sidebar-logo {
+  font-size: 3.5rem;
+  margin-top: 15px;
+  margin-bottom: 75px;
+  text-align: center;
+  font-family: 'Shadows Into Light', cursive;
+}
+
+ul.side-nav.fixed li.logo {
```
Comment (Divoolej) [inline app/styles/sidebar.scss]: Not a CSS expert here (actually I hate it a lot) but I've heard that a good practice is to never use HTML tags as selectors in your stylesheets. I guess a rule of thumb is - always use classes, so no `ul`, no `li`, no `a`. Also, you should probably always use nesting (since it's SCSS), notice that `ul.side-nav.fixed` appears twice, this could be fixed by creating a `.side-nav` selector, inside it nest a `.fixed` selector - and then `.logo` or whatever you need. All in all this file looks very good and I like the comments, too. If you'd like to become a Very Good Developer, I recommend reading about BEM (most Pros use it) - http://getbem.com/introduction/ or http://cssguidelin.es/#bem-like-naming for a quick reference. :)

File: app/templates/components/category-list.hbs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{{yield}}
+<ul class="collection">
```
Comment (Divoolej) [inline app/templates/components/category-list.hbs:2]: What do you think about naming this class `categories` or something that would tell us more about the actual content?

File: tests/integration/components/category-list-test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  integration: true
+});
+
+test('it renders', function(assert) {
+  // Set any properties with this.set('myProperty', 'value');
+  // Handle any actions with this.on('myAction', function(val) { ... });
+
+  this.render(hbs`{{category-list}}`);
+
+  assert.equal(this.$().text().trim(), '');
```
Comment (Divoolej) [inline tests/integration/components/category-list-test.js:14]: Oh wait, we actually have some tests :D Using Travis might not a bad idea after all.

**Comment by Divoolej** [comment]
It's perfect! I want it merged! But maybe @sosnowski might want to add something to the review?

File: .gitignore (hunk 1 above)
Comment (tmakuch) [inline .gitignore]: Ignore .idea instead?

Comment (TrebuhD) [inline .gitignore]: Good .idea ^^

2016-04-25 10:09 GMT+02:00 tmakuch notifications@github.com:

File: app/styles/sidebar.scss (hunk 4 above)
Comment (tmakuch) [inline app/styles/sidebar.scss]: Good catch with using element names. Point of not using them is to avoid need of rewritng styles to apply them in another case (where same html structure does not have to be used).

File: package.json [hunk 7]
```diff
@@ -19,32 +19,36 @@
...
-    "ember-cli-inject-live-reload": "^1.3.1",
-    "ember-cli-materialize": "0.18.8",
-    "ember-cli-qunit": "^1.2.1",
+    "ember-cli-inject-live-reload": "^1.4.0",
+    "ember-cli-jshint": "^1.0.0",
+    "ember-cli-materialize": "0.19.2",
+    "ember-cli-qunit": "^1.4.0",
     "ember-cli-release": "0.2.8",
-    "ember-cli-sass": "5.3.1",
+    "ember-cli-sass": "git://github.com/mike-north/ember-cli-sass.git#8ceb57d41f5774e8ececb5d1f05454449c19000c",
```
Comment (tmakuch) [inline package.json:35]: But why? Is it latest change and you're waiting for the release?

Comment (TrebuhD) [inline package.json:35]: Honestly I don't know - this was changed automatically after updating ember. Weird.

File: app/templates/components/category-list.hbs (hunk 5 above)
Comment (TrebuhD) [inline app/templates/components/category-list.hbs:2]: I thought about that, but was stopped by naming conventions: 
"Assertion Failed: You cannot use 'categories' as a component name. Component names must contain a hyphen."
